website/docs/common-scripts/xudt.mdx

```diff
@@ -0,0 +1,346 @@
+
+```yaml
+Inputs:
+    <vec> xUDT_Cell
+        Data:
+            <amount: uint128> <xUDT data>
+        Type:
+            code_hash: xUDT type script
+            args: <owner lock script hash> <xUDT args, hash of raw extension data>
+        Lock:
+            <user defined>
+    <...>
+Outputs:
+    <vec> xUDT_Cell
+        Data:
+            <amount: uint128> <xUDT data>
+        Type:
+            code_hash: xUDT type script
+            args: <owner lock script hash> <xUDT args, hash of raw extension data>
+        Lock:
+            <user defined>
+    <...>
+Witnesses:
+    WitnessArgs structure:
+        Lock: <user defined>
+        Input Type: XudtWitness
+            owner_script: <None>
+            owner_signature: <None>
+            extension_scripts:
+                <vec> ScriptVec
+                    <script>
+                <...>
+            extension_data:
+                <vec> BytesVec
+                    <data>
+                <...>
+```
+
+The only difference here is that `XudtWitness` in `input_type` or
+`output_type` field in the corresponding WitnessArgs structure, contains raw
+extension data in `ScriptVec` data structure, xUDT script must first validate
+that the hash of raw extension data provide here, is the same as blake160 hash
+included in xUDT args. After this, it uses the same logic as the previous
+workflow.
+
+### Owner Mode without Consuming Cell
+
+As described above, If an input cell uses an input lock script with same script
+hash as the owner lock script hash, the `is_owner_mode` will be set to true. It
+isn’t convenient: this requires extra cell to be consumed. With `owner_script`
+and `owner_signature` set to proper values, we can use owner mode without extra
+cell.
+
+```yaml
+Inputs:
+    <vec>
+			<Any input cells>
+    <...>
+Outputs:
+    <vec> xUDT_Cell
+        Data:
+            <amount: uint128> <xUDT data>
+        Type:
+            code_hash: xUDT type script
+            args: <owner lock script hash 1> <xUDT args>
+        Lock:
+            <user defined>
+    <...>
+Witnesses:
+    WitnessArgs structure:
+        Lock: <user defined>
+        Input Type: <None>
+        Output Type: XudtWitness
+            owner_script: <owner script 1>
+            owner_signature: <signature 1>
+            extension_scripts:
+                <vec> ScriptVec
+                    <script>
+                <...>
+            extension_data:
+                <vec> BytesVec
+                    <data>
+                <...>
+```
+
+The example above shows a scenario of owner mode without consuming the owner's
+cell. We can implement an extension script as `<owner script 1>` with signature
+validation. The `<signature 1>` can be used by `<owner script 1>` to place
+signature information.
+
+## Script Deployed Info
+
+An [implementation](https://github.com/nervosnetwork/ckb-production-scripts/blob/master/c/xudt_rce.c) of
+the spec above has been deployed to Mirana CKB mainnet and Pudge testnet:
```

Comment:
The [xUDT spec](https://github.com/nervosnetwork/ckb-production-scripts/blob/master/c/xudt_rce.c) has been deployed to CKB's Mainnet Mirana and Testnet Pudge. The following are the deployment details